src/libponyrt/gc/gc.c

```diff
@@ -591,3 +591,61 @@ void ponyint_gc_destroy(gc_t* gc)
     gc->delta = NULL;
   }
 }
+
+void pony_acquire(pony_ctx_t* ctx, void* p, pony_trace_fn f, int immutable)
+{
+  chunk_t* chunk = (chunk_t*)ponyint_pagemap_get(p);
+  pony_actor_t* actor = ponyint_heap_owner(chunk);
+  object_t* obj = NULL;
+  if (actor == ctx->current)
+  {
+    gc_t* gc = ponyint_actor_gc(ctx->current);
+    obj = ponyint_objectmap_getorput(&gc->local, p, gc->mark);
+    obj->rc++;
+  }
+  else
+  {
+    actorref_t* aref = ponyint_actormap_getorput(&ctx->acquire, actor, 0);
+    obj = ponyint_actorref_getorput(aref, p, 0);
+    obj->rc = 1;
+  }
+  if (immutable != 0)
+    obj->immutable = true;
+
+  if (!obj->immutable)
+    recurse(ctx, p, f);
+
+  ponyint_gc_handlestack(ctx);
+  ponyint_gc_sendacquire(ctx);
+}
+
+void pony_release(pony_ctx_t* ctx, void* p, pony_trace_fn f, int immutable)
+{
+  chunk_t* chunk = (chunk_t*)ponyint_pagemap_get(p);
+  pony_actor_t* actor = ponyint_heap_owner(chunk);
+  object_t* obj = NULL;
+  actorref_t* aref = NULL;
+  if (actor == ctx->current)
+  {
+    gc_t* gc = ponyint_actor_gc(ctx->current);
+    obj = ponyint_objectmap_getobject(&gc->local, p);
+    assert(obj != NULL);
+    obj->rc--;
+  }
+  else
+  {
+    aref = (actorref_t*)POOL_ALLOC(actorref_t);
+    memset(aref, 0, sizeof(actorref_t));
+    aref->actor = actor;
+    obj = ponyint_actorref_getorput(aref, p, 0);
```

Comment:
This could be `ponyint_actorref_get` instead of `getorput`, but the `get` version doesn't exist. Do we need it?